feat(storage): add validateObjectExistence option in get api (#5372)
aws-amplify/docs: src/fragments/lib/storage/js/download.mdx

  download?: boolean, // defaults to false
  expires?: number, // validity of the URL, in seconds. defaults to 900 (15 minutes)
  contentType?: string // set return content type, eg "text/html"
  validateObjectExistence?: boolean // defaults to false
});
```

> `Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this method **does not check if the file actually exists by default**. But you can check for the existence of file by setting `validateObjectExistence` to `true`.

Reviewer: Suggest updating method -> function for JS docs. Imo we can also just refer to the check as the check after talking about it in the penultimate sentence.
```suggestion
`Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this function **does not check if the file exists by default**. You can enable this check by setting `validateObjectExistence` to `true`.
```

> `Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this method **does not check if the file actually exists by default**. But you can check for the existence of file by setting `validateObjectExistence` to `true`.

Reviewer 2: why is it our recommended option?

Author: This is a good question. If we have set download to true it would return complete Blob object in the API call response. In order to avoid loading complete object with get call we recommend setting download flag to false. please let me know if have any concerns.

Reviewer 2: I think we need to add a bit of a description on why that is our recommended option. what do you think @jimblanc and @stocaaro ?

> `Storage.get` returns a signed URL `string` to your file, if `download` is false, which is the default. You can use this to create a download link for people to click on. This is our recommended option. Note that this method **does not check if the file actually exists by default**. But you can check for the existence of file by setting `validateObjectExistence` to `true`.

Reviewer 3: Suggested changes to this line with some changes to the existing wording, please double check me.

```suggestion
`Storage.get` returns a signed URL `string` to your file if `download` is false, which is the default. You can use this to create a download link for users to click on. Note that this method **does not check if the file actually exists by default**. You can check that the file exists when generating a signed URL by setting `validateObjectExistence` to `true`.
```

const result = await Storage.get(`filename.txt`, { download: true });

// data.Body is a Blob
result.Body.text().then(string => { 
  // handle the String data return String 
result.Body.text().then((string) => {
  // handle the String data return String
});
```

> Note that the `Blob` methods like `.text()` are not supported on [IE/Opera/Safari](https://developer.mozilla.org/en-US/docs/Web/API/Blob/text); in those cases you can [parse manually](https://developer.mozilla.org/en-US/docs/Web/API/Blob#JavaScript).

Reviewer 3: Just to double check, this is being removed because it's not longer relevant for our supported browsers?

Author: Yes, Correct currently .text() method is being supported in the browsers.

await Storage.get('filename.txt', { expires: 60 });
```

> ### Check existence of file

Reviewer: Nit: Arguable if this is better but it feels a bit more like treating the existence as a noun?
```suggestion
### Check for existence of a file
```

Author: Thanks for the suggestion.

> ### Check existence of file

Reviewer 2: Is there a reason why we are calling this out again in the guide?

Reviewer 3: +1, I think the change at the top of the page is adequate

Author: Updated. Thanks.

> You can check the existence of file in the storage using `validateObjectExistence` option. If the the file is existed the get call returns a pre-signed URL and if the file does not exist then return a 404 Error.

Reviewer: Nit:
```suggestion
You can check for the existence of a file in the storage using the `validateObjectExistence` option. When enabled a `get` call will return a pre-signed URL if the file exists and a `404` error if it does not.
```

> You can check for the existence of a file in the storage using the `validateObjectExistence` option. When enabled a `get` call will return a pre-signed URL if the file exists and a `404` error if it does not.

Reviewer 4: ```suggestion
You can check for the existence of a file in the storage category's `get` method using the `validateObjectExistence` option. When this flag is enabled a `get` call will return a pre-signed URL if the file exists and a `404` error if it does not.
```

```javascript

> // To check existence of file

Reviewer: Same nit:
```suggestion
// To check for existence of a file
```
